src/ng/cookieReader.js

```diff
@@ -22,6 +22,13 @@ function $$CookieReader($document) {
     }
   }
 
+  //null object implementation in case document was mocked out
+  if (!rawDocument || typeof rawDocument.cookie !== 'string') {
+    return function() {
+      return lastCookies;
+    };
+  }
+
```

Comment:
actually this is the only place that breaks, you can mock out `$window` and not include `$document`. See https://github.com/angular/angular.js/blob/master/test/ng/windowSpec.js